posts/renv-complications/index.qmd

```diff
@@ -0,0 +1,131 @@
+---
+title: "Things that can go wrong when using renv"
+author:
+  - name: "Hugo Gruson"
+    orcid: "0000-0002-4094-1476"
+date: "2023-09-26"
+categories: [R, reproducibility, renv]
+format:
+  html: 
+    toc: true
+---
+
+We use the [renv R package](https://rstudio.github.io/renv/) to ensure reproducibility of the training materials and the pipelines we are providing.
+But we sometimes get reports from users who struggle to rebuild the environment and run the code.
+
+In this post, we dissect the source of these issues, explain why in reality renv is not at fault, and how this is caused by the inherent complexity of reproducibility. renv documentation already includes [caveats](https://rstudio.github.io/renv/articles/renv.html#caveats) explaining why some situations are bound to require more complex tools. This blog post reiterates some of these caveats and illustrates them with concrete examples.
+
+We finish by mentioning a couple of more complete (but more complex!) frameworks that can overcome the issues presented here.
+We do not explore these alternative framework in details but we provide links to more information.
+
+## Binaries vs building from source
+
+Software, including R packages, can generally be delivered in two forms: as binaries or as source code.
+If you are installing as source code, you may in some case need a compilation toolchain on your computer.
+If the toolchain is missing, it can lead to errors such as:
+
+> ```txt
+> ld: warning: search path '/opt/gfortran/lib' not found
+> ld: library 'gfortran' not found
+> ```
+
+Most of the time, regular users of R will not see these errors because they are installing binaries.
+Indeed, CRAN provides pre-compiled binaries for Windows and macOS for the last version of the package and R.
+
+With renv, you often want to install older versions of the packages, which won't be available as binaries from CRAN.
+This means you are more likely to have to compile the package yourself and see this kind of errors, even though renv is not causing them.
+
+## Beyond renv scope: incompatibility with system dependency versions
+
+We [discussed previously the topic of system dependencies](../system-dependencies/), and [dependencies on specific R versions](https://blog.r-hub.io/2022/09/12/r-dependency/). These special dependencies can also be a source of headaches when using renv.
+
+The heart of the issue is that renv provides a simplified solution to reproducibility: it focuses on R packages and R package versions.
```

Comment:
```suggestion
The heart of the issue is that renv provides a simplified solution to reproducibility: it focuses on R packages and their versions.
```

In reference to my other comment on https://github.com/epiverse-trace/epiverse-trace.github.io/pull/154#discussion_r1466375927 - I think I read R versions here 😅